apps/prairielearn/src/tests/activeAccessRestriction.test.ts

```diff
@@ -152,11 +155,25 @@ describe('Exam and homework assessment with active access restriction', function
     assert.include(examInstanceUrl, '/assessment_instance/');
     context.examInstanceUrl = examInstanceUrl;
 
-    // save the examQuestionUrl for later
-    const examQuestionUrl = response.$('a:contains("Question 1")').attr('href');
-    context.examQuestionUrl = `${context.siteUrl}${examQuestionUrl}`;
-
+    // Save context for future tests
     context.__csrf_token = response.$('span[id=test_csrf_token]').text();
+    const questionWithVariantPath = response.$('a:contains("Question 1")').attr('href');
+    const questionWithoutVariantPath = response.$('a:contains("Question 2")').attr('href');
+    context.examQuestionUrl = `${context.siteUrl}${questionWithVariantPath}`;
+    context.examQuestionWithoutVariantUrl = `${context.siteUrl}${questionWithoutVariantPath}`;
+
+    const questionResponse = await helperClient.fetchCheerio(context.examQuestionUrl, {
+      headers,
+    });
+    assert.isTrue(questionResponse.ok);
+  });
+
+  step('count number of variants generated', async () => {
+    const countVariantsResult = await sqldb.queryOneRowAsync(sql.count_variants, {
+      assessment_instance_id: helperClient.parseAssessmentInstanceId(context.examInstanceUrl),
+    });
+    context.numberOfVariants = countVariantsResult.rows[0].count;
```

Comment:
Nit, I'd prefer to use Zod-validated queries both here and elsewhere in the file (Prettier formatting of the below will be necessary).

```suggestion
    context.numberOfVariants = await sqldb.queryRow(sql.count_variants, {
      assessment_instance_id: helperClient.parseAssessmentInstanceId(context.examInstanceUrl),
    }, z.number());
```

Reply:
This query actually performs a `COUNT`, which IIRC results in a bigint, so `z.number()` will not work. This will either need to be `z.coerce.number()` or the query will need to be updated to `COUNT(v.*)::INT`.

Reply:
Good point, `::integer` is the way we usually handle that.

Reply:
Done